src/browser/web-view.js

```diff
@@ -919,18 +929,19 @@ const decodeError = compose(ReportError, decodeDetail);
 
 // Navigation
 
-const decodeLocationChange = ({detail}) => {
-  var uri;
+const decodeLocationChange = ({detail}) =>
   // Servo and Gecko have different implementation of detail.
   // In Gecko, detail is a string (the uri).
   // In Servo, detail is an object {uri,canGoBack,canGoForward}
-  if (typeof detail == 'string') {
-    uri = detail;
-  } else {
-    uri = detail.uri;
-  }
-  return LocationChanged(uri, performance.now());
-}
+  ( typeof detail === 'string'
+  ? LocationChanged(detail, false, false, performance.now())
```

Comment:
This seems little odd. Usually when location changes that means that you can actually go back because you navigated from place `A` to place `B`. I would suggest to instead make types of the `canGoBack` and `canGoForward` fields `?boolean` and set those to `null` in gecko case. So we only modify `navigation.canGoBack` / `navigation.canGoForward` when action has boolean values for corresponding fields.